Source/AST/Nodes/Node.swift

```diff
@@ -16,6 +16,9 @@ public protocol Node {
     
     /// The wrapped child nodes.
     var children: [Node] { get }
+
+    /// Sequence of wrapped child nodes.
+    func childSequence() -> ChildSequence
```

Comment:
I'm curious what is the motivation for exposing this is. To allow lazy enumeration?

Reply:
Perhaps we don't need to declare it as a requirement on the protocol because this would require a major version bump. Instead I don't see why we can't provide a computed property in an extension:

```swift
public extension Node {
  
  var childSequence: ChildSequence {
    return ChildSequence(node: cmarkNode)
  }

}
```

Reply:
Hm, never considered the versioning issues with that. Moving this to a protocol extension without the requirement will work. Client code couldn’t provide a custom implementation of this requirements anyways since the ChildSequence has no public initializer.